packages/react-native-worklets/Common/cpp/worklets/SharedItems/Shareables.cpp

```diff
@@ -90,17 +85,17 @@ jsi::Value makeShareableClone(
       }
     }
   } else if (value.isString()) {
-    shareable = std::make_shared<ShareableString>(value.asString(rt).utf8(rt));
+    return makeShareableString(rt, value.asString(rt));
   } else if (value.isUndefined()) {
-    shareable = std::make_shared<ShareableScalar>();
+    return makeShareableUndefined(rt);
   } else if (value.isNull()) {
-    shareable = std::make_shared<ShareableScalar>(nullptr);
+    return makeShareableNull(rt);
   } else if (value.isBool()) {
-    shareable = std::make_shared<ShareableScalar>(value.getBool());
+    return makeShareableBoolean(rt, value.getBool());
   } else if (value.isNumber()) {
-    shareable = std::make_shared<ShareableScalar>(value.getNumber());
+    return makeShareableNumber(rt, value.getNumber());
   } else if (value.isBigInt()) {
-    shareable = std::make_shared<ShareableBigInt>(rt, value.getBigInt(rt));
+    return makeShareableBigInt(rt, value.getBigInt(rt));
```

Comment:
Shouldn't this be done in a separate PR?

Reply:
the code is the same but I can do it in a separate PR